<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Knot_Window</class>
 <widget class="QMainWindow" name="Knot_Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Knotter</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="extended_TabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>-1</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideLeft</enum>
      </property>
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menuOpen_Recent">
     <property name="title">
      <string>Open &amp;Recent</string>
     </property>
     <addaction name="actionNo_recent_files"/>
    </widget>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="menuOpen_Recent"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="actionSave_As"/>
    <addaction name="actionSave_A_ll"/>
    <addaction name="action_Export"/>
    <addaction name="separator"/>
    <addaction name="action_Print"/>
    <addaction name="actionPage_Set_up"/>
    <addaction name="actionPr_int_Preview"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menuNo_des">
    <property name="title">
     <string>No&amp;des</string>
    </property>
    <addaction name="action_Link"/>
    <addaction name="action_Unlink"/>
    <addaction name="action_Erase"/>
    <addaction name="action_Merge"/>
    <addaction name="separator"/>
    <addaction name="action_Horizontal_Flip"/>
    <addaction name="action_Vertical_Flip"/>
    <addaction name="action_Rotate"/>
    <addaction name="action_Scale"/>
    <addaction name="separator"/>
    <addaction name="actionS_nap_to_grid"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="action_Undo"/>
    <addaction name="action_Redo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="action_Copy"/>
    <addaction name="action_Paste"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_All"/>
    <addaction name="separator"/>
    <addaction name="action_Preferences"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <widget class="QMenu" name="menu_Zoom">
     <property name="title">
      <string>&amp;Zoom</string>
     </property>
     <addaction name="actionZoom_In"/>
     <addaction name="actionZoom_Out"/>
     <addaction name="action_Reset_Zoom"/>
    </widget>
    <widget class="QMenu" name="menu_Toolbars">
     <property name="title">
      <string>&amp;Toolbars</string>
     </property>
    </widget>
    <widget class="QMenu" name="menu_Docks">
     <property name="title">
      <string>&amp;Docks</string>
     </property>
    </widget>
    <widget class="QMenu" name="menuR_endering">
     <property name="title">
      <string>R&amp;endering</string>
     </property>
     <addaction name="action_Normal_Rendering"/>
     <addaction name="action_Highlight_loops"/>
     <addaction name="action_Simplified"/>
    </widget>
    <addaction name="menu_Toolbars"/>
    <addaction name="menu_Docks"/>
    <addaction name="separator"/>
    <addaction name="menu_Zoom"/>
    <addaction name="action_Reset_View"/>
    <addaction name="separator"/>
    <addaction name="actionShow_Graph"/>
    <addaction name="actionShow_Knot_line"/>
    <addaction name="menuR_endering"/>
    <addaction name="separator"/>
    <addaction name="actionEnable_Grid"/>
    <addaction name="actionConfigure_G_rid"/>
    <addaction name="action_Move_Grid"/>
    <addaction name="separator"/>
    <addaction name="action_Background"/>
   </widget>
   <widget class="QMenu" name="menu_Knot">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="actionInsert_Edges"/>
    <addaction name="actionEdge_List"/>
    <addaction name="separator"/>
    <addaction name="actionInsert_Polygon"/>
    <addaction name="separator"/>
    <addaction name="actionRefresh_Path"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_Manual"/>
    <addaction name="separator"/>
    <addaction name="action_About"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
   <addaction name="menu_View"/>
   <addaction name="menuNo_des"/>
   <addaction name="menu_Knot"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="EditBar">
   <property name="windowTitle">
    <string>Knot Editing</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllToolBarAreas</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionInsert_Edges"/>
   <addaction name="actionEdge_List"/>
   <addaction name="separator"/>
   <addaction name="action_Link"/>
   <addaction name="action_Unlink"/>
   <addaction name="action_Erase"/>
   <addaction name="action_Merge"/>
   <addaction name="separator"/>
   <addaction name="actionInsert_Polygon"/>
  </widget>
  <widget class="QToolBar" name="MainToolBar">
   <property name="windowTitle">
    <string>Main</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
   <addaction name="actionSave_As"/>
   <addaction name="action_Export"/>
   <addaction name="separator"/>
   <addaction name="action_Undo"/>
   <addaction name="action_Redo"/>
   <addaction name="separator"/>
   <addaction name="actionCut"/>
   <addaction name="action_Copy"/>
   <addaction name="action_Paste"/>
  </widget>
  <widget class="QToolBar" name="ViewBar">
   <property name="windowTitle">
    <string>View</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
   <addaction name="actionZoom_In"/>
   <addaction name="actionZoom_Out"/>
   <addaction name="separator"/>
   <addaction name="actionShow_Graph"/>
   <addaction name="actionEnable_Grid"/>
   <addaction name="actionS_nap_to_grid"/>
  </widget>
  <widget class="QToolBar" name="TransformBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Horizontal_Flip"/>
   <addaction name="action_Vertical_Flip"/>
   <addaction name="separator"/>
   <addaction name="action_Rotate"/>
   <addaction name="action_Scale"/>
  </widget>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save &amp;As...</string>
   </property>
  </action>
  <action name="action_Export">
   <property name="text">
    <string>E&amp;xport...</string>
   </property>
  </action>
  <action name="action_Open">
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="action_New">
   <property name="text">
    <string>&amp;New</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionInsert_Edges">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Edit graph</string>
   </property>
   <property name="toolTip">
    <string>Edit edges and nodes</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+E</string>
   </property>
  </action>
  <action name="actionEdge_List">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Edge &amp;List</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+L</string>
   </property>
  </action>
  <action name="action_Link">
   <property name="text">
    <string>&amp;Link</string>
   </property>
   <property name="toolTip">
    <string>Link Selected Nodes</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="action_Erase">
   <property name="text">
    <string>&amp;Erase</string>
   </property>
   <property name="toolTip">
    <string>Erase Selected Nodes</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_Merge">
   <property name="text">
    <string>&amp;Merge</string>
   </property>
   <property name="toolTip">
    <string>Merge Selected Nodes</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="action_Undo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="action_Redo">
   <property name="text">
    <string>&amp;Redo</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="text">
    <string>Select &amp;All</string>
   </property>
  </action>
  <action name="actionRefresh_Path">
   <property name="text">
    <string>Refresh &amp;Path</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom &amp;In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom &amp;Out</string>
   </property>
  </action>
  <action name="action_Reset_Zoom">
   <property name="text">
    <string>&amp;Reset Zoom</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="action_Reset_View">
   <property name="text">
    <string>&amp;Reset View</string>
   </property>
  </action>
  <action name="actionShow_Graph">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show &amp;Graph</string>
   </property>
  </action>
  <action name="actionShow_Knot_line">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Knot&amp;line</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cu&amp;t</string>
   </property>
  </action>
  <action name="action_Copy">
   <property name="text">
    <string>&amp;Copy</string>
   </property>
  </action>
  <action name="action_Paste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
  </action>
  <action name="action_Preferences">
   <property name="text">
    <string>Pre&amp;ferences...</string>
   </property>
  </action>
  <action name="actionEnable_Grid">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Enable Grid</string>
   </property>
  </action>
  <action name="actionConfigure_G_rid">
   <property name="text">
    <string>Configure G&amp;rid...</string>
   </property>
  </action>
  <action name="action_Horizontal_Flip">
   <property name="text">
    <string>&amp;Horizontal Flip</string>
   </property>
  </action>
  <action name="action_Vertical_Flip">
   <property name="text">
    <string>&amp;Vertical Flip</string>
   </property>
  </action>
  <action name="action_Move_Grid">
   <property name="text">
    <string>&amp;Move Grid</string>
   </property>
  </action>
  <action name="actionNo_recent_files">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>No recent files</string>
   </property>
  </action>
  <action name="action_Manual">
   <property name="text">
    <string>&amp;Manual</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="action_Unlink">
   <property name="text">
    <string>&amp;Unlink</string>
   </property>
   <property name="toolTip">
    <string>Unlink selected nodes</string>
   </property>
  </action>
  <action name="actionInsert_Polygon">
   <property name="text">
    <string>Insert Pol&amp;ygon...</string>
   </property>
  </action>
  <action name="action_Rotate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Rotate</string>
   </property>
  </action>
  <action name="action_Scale">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Scale</string>
   </property>
  </action>
  <action name="action_Background">
   <property name="text">
    <string>&amp;Background...</string>
   </property>
  </action>
  <action name="actionSave_A_ll">
   <property name="text">
    <string>Save A&amp;ll</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionS_nap_to_grid">
   <property name="text">
    <string>S&amp;nap to grid</string>
   </property>
  </action>
  <action name="action_Print">
   <property name="text">
    <string>&amp;Print...</string>
   </property>
  </action>
  <action name="actionPage_Set_up">
   <property name="text">
    <string>Page Set&amp;up...</string>
   </property>
  </action>
  <action name="actionPr_int_Preview">
   <property name="text">
    <string>Pr&amp;int Preview...</string>
   </property>
  </action>
  <action name="action_Highlight_loops">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Highlight loops</string>
   </property>
  </action>
  <action name="action_Normal_Rendering">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Normal</string>
   </property>
   <property name="toolTip">
    <string>Full Rendering</string>
   </property>
  </action>
  <action name="action_Simplified">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Simplified</string>
   </property>
   <property name="toolTip">
    <string>Simplified rendering (faster)</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>extended_TabWidget</class>
   <extends>QTabWidget</extends>
   <header>extended_tabwidget.hpp</header>
   <container>1</container>
   <slots>
    <signal>drag_away(int)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>action_New</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Save</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Open</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Export</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>export_image()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_In</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>zoom_in()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_Out</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>zoom_out()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCut</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Copy</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Paste</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConfigure_G_rid</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>configure_grid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEnable_Grid</sender>
   <signal>triggered(bool)</signal>
   <receiver>Knot_Window</receiver>
   <slot>enable_grid(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Preferences</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>config()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Manual</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>show_help()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>drag_away(int)</signal>
   <receiver>Knot_Window</receiver>
   <slot>drag_tab_away(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>-9</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>mode_edge_list()</slot>
  <slot>mode_edge()</slot>
  <slot>mode_node()</slot>
  <slot>mouse_moved(QPointF)</slot>
  <slot>clear()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
  <slot>open()</slot>
  <slot>export_image()</slot>
  <slot>zoom_in()</slot>
  <slot>zoom_out()</slot>
  <slot>copy()</slot>
  <slot>cut()</slot>
  <slot>paste()</slot>
  <slot>enable_grid(bool)</slot>
  <slot>configure_grid()</slot>
  <slot>cause_crash()</slot>
  <slot>config()</slot>
  <slot>show_help()</slot>
  <slot>drag_tab_away(int)</slot>
 </slots>
</ui>
